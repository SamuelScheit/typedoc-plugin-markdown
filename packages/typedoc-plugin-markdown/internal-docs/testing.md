---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
