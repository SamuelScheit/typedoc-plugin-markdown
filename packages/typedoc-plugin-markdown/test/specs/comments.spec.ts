import { expectFileToEqual } from '@devtools/testing';

describe(`Comments`, () => {
  test(`should compile comments for module`, () => {
    expectFileToEqual('comments', ['members'], ['README.md']);
  });

  test(`should handle single example tags`, () => {
    expectFileToEqual(
      'comments',
      ['members'],
      ['/functions/multipleExampleTags.md'],
    );
  });

  test(`should handle multiple example tags`, () => {
    expectFileToEqual(
      'comments',
      ['members'],
      ['/functions/multipleExampleTags.md'],
    );
  });

  test(`should get tables for properties`, () => {
    expectFileToEqual('comments', 'members', [
      '/classes/ClassPropertiesTable.md',
      '/Class.ClassPropertiesTable.md',
    ]);
  });

  test(`should get tables for parameters`, () => {
    expectFileToEqual('comments', 'members', ['/functions/parametersTable.md']);
  });

  test(`should get tables for type declarations`, () => {
    expectFileToEqual('comments', 'members', [
      '/type-aliases/TypeDeclarationTable.md',
      '/TypeAlias.TypeDeclarationTable.md',
    ]);
    expectFileToEqual('comments', 'members', [
      '/variables/TypeDeclarationTable.md',
      '/Variable.TypeDeclarationTable.md',
    ]);
  });

  test(`should get tables for emum`, () => {
    expectFileToEqual('comments', 'members', [
      '/enumerations/EnumMembersTable.md',
      '/Enumeration.EnumMembersTable.md',
    ]);
  });

  test(`should get function with block tags`, () => {
    expectFileToEqual('comments', 'members', [
      '/functions/functionWithBlockTags.md',
      '/Function.functionWithBlockTags.md',
    ]);
  });

  test(`should get variable with block tags`, () => {
    expectFileToEqual('comments', 'members', [
      '/type-aliases/typeWithBlockTags.md',
      '/TypeAlias.typeWithBlockTags.md',
    ]);
  });
});
