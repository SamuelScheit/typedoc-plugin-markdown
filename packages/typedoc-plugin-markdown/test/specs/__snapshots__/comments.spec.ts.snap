// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should compile comments for module: (Output File Strategy "members") (Option Group "1") 1`] = `
"# typedoc-stubs

Comments form module comments

## Links

Links using \`{@link}\` inline tags.

- [CommentInterface](interfaces/CommentInterface.md) - Links to CommentInterface
- [Links to CommentInterface.prop](interfaces/CommentInterface.md)
- [Links to CommentInterface.propb](interfaces/CommentInterface.md)
- [CommentEnum.MemberB](enumerations/CommentEnum.md)
- [SameName:var](variables/SameName.md)
- [SameName:interface](interfaces/SameName.md)
- [SameName.prop](interfaces/SameName.md)
- [prop:var](variables/prop.md)

External links:

- [Google](https://www.google.com)
- [\`https://www.google.com\`](https://www.google.com)

## Tag A

Comments for a tag

## Tag B

Comments for tag written on same line

## Html And Jsx

A \`<tag>\` \`</tag>\`  in comments

Closing \`</element>\`  on its own

A \`<tag>\` in backticks

\`\`\`
A <tag> in a code block
Some <p> html </p> inside codeblock
\`\`\`

## Enumerations

- [CommentEnum](enumerations/CommentEnum.md)

## Interfaces

- [CommentInterface](interfaces/CommentInterface.md)
- [SameName](interfaces/SameName.md)

## Variables

- [SameName](variables/SameName.md)
- [prop](variables/prop.md)
- [propb](variables/propb.md)
"
`;

exports[`Comments should compile comments for module: (Output File Strategy "members") (Option Group "2") 1`] = `
"# typedoc-stubs

Comments form module comments

## Links

Links using \`{@link}\` inline tags.

- [CommentInterface](interfaces/CommentInterface.md) - Links to CommentInterface
- [Links to CommentInterface.prop](interfaces/CommentInterface.md#prop)
- [Links to CommentInterface.propb](interfaces/CommentInterface.md#propb)
- [CommentEnum.MemberB](enumerations/CommentEnum.md#memberb)
- [SameName:var](variables/SameName.md)
- [SameName:interface](interfaces/SameName.md)
- [SameName.prop](interfaces/SameName.md#prop)
- [prop:var](variables/prop.md)

External links:

- [Google](https://www.google.com)
- [\`https://www.google.com\`](https://www.google.com)

## Tag A

Comments for a tag

## Tag B

Comments for tag written on same line

## Html And Jsx

A \`<tag>\` \`</tag>\`  in comments

Closing \`</element>\`  on its own

A \`<tag>\` in backticks

\`\`\`
A <tag> in a code block
Some <p> html </p> inside codeblock
\`\`\`

## Enumerations

- [CommentEnum](enumerations/CommentEnum.md)

## Interfaces

- [CommentInterface](interfaces/CommentInterface.md)
- [SameName](interfaces/SameName.md)

## Variables

- [SameName](variables/SameName.md)
- [prop](variables/prop.md)
- [propb](variables/propb.md)
"
`;

exports[`Comments should compile comments for module: (Output File Strategy "modules") (Option Group "1") 1`] = `
"# typedoc-stubs

Comments form module comments

## Links

Links using \`{@link}\` inline tags.

- [CommentInterface](README.md#commentinterface) - Links to CommentInterface
- [Links to CommentInterface.prop](README.md#commentinterface)
- [Links to CommentInterface.propb](README.md#commentinterface)
- [CommentEnum.MemberB](README.md#commentenum)
- [SameName:var](README.md#samename-1)
- [SameName:interface](README.md#samename)
- [SameName.prop](README.md#samename)
- [prop:var](README.md#prop)

External links:

- [Google](https://www.google.com)
- [\`https://www.google.com\`](https://www.google.com)

## Tag A

Comments for a tag

## Tag B

Comments for tag written on same line

## Html And Jsx

A \`<tag>\` \`</tag>\`  in comments

Closing \`</element>\`  on its own

A \`<tag>\` in backticks

\`\`\`
A <tag> in a code block
Some <p> html </p> inside codeblock
\`\`\`

## Contents

- [Enumerations](README.md#enumerations)
    - [CommentEnum](README.md#commentenum)
- [Interfaces](README.md#interfaces)
    - [CommentInterface](README.md#commentinterface)
    - [SameName](README.md#samename)
- [Variables](README.md#variables)
    - [SameName](README.md#samename-1)
    - [prop](README.md#prop)
    - [propb](README.md#propb)

## Enumerations

### CommentEnum

#### Enumeration Members

| Enumeration Member | Value | Description |
| :------ | :------ | :------ |
| \`Member\` | \`0\` | Comment for Member<br /><br />**Deprecated**<br />Deprecated member<br /><br />**See**<br />[SameName](README.md#samename-1) |
| \`MemberB\` | \`1\` | - |

## Interfaces

### CommentInterface

#### Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| \`prop\` | \`string\` | - |
| \`propb\` | \`string\` | - |

***

### SameName

#### Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| \`prop\` | \`string\` | - |
| \`propb\` | \`string\` | - |

## Variables

### SameName

> **SameName**: \`true\`

***

### prop

> **\`const\`** **prop**: \`true\` = \`true\`

***

### propb

> **\`const\`** **propb**: \`true\` = \`true\`
"
`;

exports[`Comments should compile comments for module: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# typedoc-stubs

Comments form module comments

## Links

Links using \`{@link}\` inline tags.

- [CommentInterface](README.md#commentinterface) - Links to CommentInterface
- [Links to CommentInterface.prop](README.md#prop)
- [Links to CommentInterface.propb](README.md#propb)
- [CommentEnum.MemberB](README.md#memberb)
- [SameName:var](README.md#samename-1)
- [SameName:interface](README.md#samename)
- [SameName.prop](README.md#prop-1)
- [prop:var](README.md#prop-2)

External links:

- [Google](https://www.google.com)
- [\`https://www.google.com\`](https://www.google.com)

## Tag A

Comments for a tag

## Tag B

Comments for tag written on same line

## Html And Jsx

A \`<tag>\` \`</tag>\`  in comments

Closing \`</element>\`  on its own

A \`<tag>\` in backticks

\`\`\`
A <tag> in a code block
Some <p> html </p> inside codeblock
\`\`\`

## Contents

- [Enumerations](README.md#enumerations)
    - [CommentEnum](README.md#commentenum)
- [Interfaces](README.md#interfaces)
    - [CommentInterface](README.md#commentinterface)
    - [SameName](README.md#samename)
- [Variables](README.md#variables)
    - [SameName](README.md#samename-1)
    - [prop](README.md#prop-2)
    - [propb](README.md#propb-2)

## Enumerations

### CommentEnum

#### Enumeration Members

| Enumeration Member | Value | Description |
| :------ | :------ | :------ |
| <a id="member" name="member"></a> \`Member\` | \`0\` | Comment for Member<br /><br />**Deprecated**<br />Deprecated member<br /><br />**See**<br />[SameName](README.md#samename-1) |
| <a id="memberb" name="memberb"></a> \`MemberB\` | \`1\` | - |

## Interfaces

### CommentInterface

#### Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| <a id="prop" name="prop"></a> \`prop\` | \`string\` | - |
| <a id="propb" name="propb"></a> \`propb\` | \`string\` | - |

***

### SameName

#### Properties

| Property | Type | Description |
| :------ | :------ | :------ |
| <a id="prop-1" name="prop-1"></a> \`prop\` | \`string\` | - |
| <a id="propb-1" name="propb-1"></a> \`propb\` | \`string\` | - |

## Variables

### SameName

> **SameName**: \`true\`

***

### prop

> **\`const\`** **prop**: \`true\` = \`true\`

***

### propb

> **\`const\`** **propb**: \`true\` = \`true\`
"
`;
